[DOCS] CSV settings (#161189)
elastic/kibana: docs/settings/reporting-settings.asciidoc

==== CSV settings

[[xpack-reporting-csv]] `xpack.reporting.csv.maxSizeBytes` {ess-icon}::

> The maximum {byte-units}[byte size] of a CSV file before being truncated. This setting exists to prevent large exports from causing performance and storage issues. Can be specified as number of bytes. Defaults to `250mb`.

Reviewer: Based on https://github.com/elastic/kibana/issues/159514, we haven't actually updated the default limit yet. I'd remove this for now.

Reviewer 2: I'm at a point where I can fast-track this change, as well as making `xpack.reporting.csv.scroll.duration` to be a user-editable setting in ESS. I will update here, in case the change is made while this PR is still open.

Reviewer 2: The default for maxSizeBytes will be increased in 8.10: https://github.com/elastic/kibana/pull/161318

[NOTE]
============

> We recommend using CSV reports to export moderate amounts of data only. The feature enables analysis of data in external tools, but is not designed for bulk export or to backup {es} data. Setting `xpack.reporting.csv.maxSizeBytes` larger than the default limit hat the potential to negatively affect the performance of your {es} cluster. 

Reviewer: Let's be specific and state what we mean by "moderate" here.

Minor grammar nit. We'll also want to revise "default limit" here as it hasn't actually changed yet.

```suggestion
We recommend using CSV reports to export moderate amounts of data only. The feature enables analysis of data in external tools, but it's not intended for bulk export or to backup {es} data. Setting `xpack.reporting.csv.maxSizeBytes` to larger than 250 MB can negatively affect the performance of your {es} cluster. 
```

> The following deployment configurations may lead to failed report jobs or incomplete reports:

Reviewer: It's not immediately clear to me how this relates to the previous paragraph. Can you clearly link this to CSV settings for the user?

* Any shard needed for search is unavailable.
* Data is stored on slow storage tiers. 
* Network latency between nodes is high.

> * Cross-cluster search is used. 

Reviewer: ```suggestion
* {ccs-cap} is used. 
```

> To export large amounts of data we recommend using {es} APIs directly, or Logstash CSV export. See {ref}/scroll-api.html[Scroll API], {ref}/point-in-time-api.html[Point in time API], and {ref}/sql-rest-format.html#_csv[CSV response format].

Reviewer: Let's narrow down the links to one "happy path" for the user. They shouldn't have to weed through three options.

I also don't see a link for the Logstash CSV option. Instead, it looks like you link to the CSV response format for ES SQL.... which I don't think would be appropriate for this use case. I'd also confirm that the APIs you've linked are recommended.
```suggestion
To export large amounts of data, we recommend using {es} APIs or a Logstash CSV export. See {ref}/scroll-api.html[Scroll API], {ref}/point-in-time-api.html[Point in time API], and {ref}/sql-rest-format.html#_csv[CSV response format].
```

============

`xpack.reporting.csv.scroll.size`::
Number of documents retrieved from {es} for each scroll iteration during a CSV export. Defaults to `500`.
[NOTE]
============

> If the default size of 500 covers a large number of shards in {es}, you may need to lower this setting. 

Reviewer: What is "a large number of shards?"

Reviewer 2: Based on anecdotal evidence, it starts in the range of 50-100.

============

`xpack.reporting.csv.scroll.duration`::
 Amount of {time-units}[time] allowed before {kib} cleans the scroll context during a CSV export. Defaults to `30s`.
[NOTE]
============

> If search latency in {es} is sufficiently high, such as if you are using cross-cluster search, you may need to increase the setting. 

Reviewer: ```suggestion
If search latency in {es} is sufficiently high, such as if you are using {ccs}, you may need to increase the setting. 
```

Reviewer 2: Other than cross-cluster search, using frozen tiers is another example where an admin may need to increase this setting.

Escape formula values in cells with a `'`. See OWASP: https://www.owasp.org/index.php/CSV_Injection. Defaults to `true`.

`xpack.reporting.csv.enablePanelActionDownload`::
deprecated:[7.9.0,This setting has no effect.] Enables CSV export from a saved search on a dashboard. This action is available in the dashboard panel menu for the saved search. 
[NOTE]
============

> This setting exists for backwards compatibility, but is unused and hardcoded to `true`. CSV export from a saved search on a dashboard is enabled when Reporting is enabled.

Reviewer: ```suggestion
This setting exists for backwards compatibility and is hardcoded to `true`. CSV export from a saved search on a dashboard is enabled when Reporting is enabled.
```
